endOfEpoch/metachain/pendingMiniBlocks.go

```diff
@@ -14,14 +15,18 @@ import (
 
 //ArgsPendingMiniBlocks is structure that contain components that are used to create a new pendingMiniBlockHeaders object
 type ArgsPendingMiniBlocks struct {
-	Marshalizer marshal.Marshalizer
-	Storage     storage.Storer
+	Marshalizer      marshal.Marshalizer
+	Storage          storage.Storer
+	MetaBlockStorage storage.Storer
+	MetaBlockPool    storage.Cacher
 }
 
 type pendingMiniBlockHeaders struct {
-	marshalizer marshal.Marshalizer
-	storage     storage.Storer
+	marshalizer      marshal.Marshalizer
+	metaBlockStorage storage.Storer
+	metaBlockPool    storage.Cacher
 
+	storage             storage.Storer
 	mutPending          sync.Mutex
 	mapMiniBlockHeaders map[string]block.ShardMiniBlockHeader
```

Comment:
Why you not use map[string]*block.ShardMiniBlockHeader

Reply:
it is needed in a direct way. It will be copied to the new metablock